lib/main.ts

```diff
@@ -296,6 +293,10 @@ export class Transpiler {
       // code is not a generic compiler, so only yields TS errors if they could
       // be the cause of facade generation issues.
       // This greatly speeds up tests and execution.
+
+      // TODO(derekx): This has gotten slower since upgrading from TS 1.7 to TS 3. Removing the
+      // following line makes the tests finish 2 seconds faster. Investigate whether or not there's
+      // a way to speed it up.
       diagnostics = diagnostics.concat(program.getSemanticDiagnostics());
```

Comment:
do we need the semantics diagnostics?

Reply:
It says above that the semantic diagnostics only get displayed when the facade generation error could be due to TS code errors, so it would probably be good to keep. Maybe we could prevent the tests from running semantic diagnostics though.

Reply:
Maybe add a flag that turns on and off the diagnostics and leave it off by default for the tests. Having a fast test cycle is really important to be productive making changes to this codebase.